lib/web_ui/dev/edge_installation.dart

```diff
@@ -68,12 +71,73 @@ Future<BrowserInstallation> getEdgeInstallation(
     // Since Edge is included in Windows, always assume there will be one on the
     // system.
     infoLog.writeln('Using the system version that is already installed.');
+    final EdgeLauncher edgeLauncher = EdgeLauncher();
+    if (edgeLauncher.isInstalled) {
+      infoLog.writeln('Launcher installation was skipped, already installed.');
+    } else {
+      infoLog.writeln('Installing MicrosoftEdgeLauncher');
+      await edgeLauncher.install();
+      infoLog.writeln(
+          'Installations complete. To launch it run ${edgeLauncher.executable}');
+    }
+
     return BrowserInstallation(
       version: 'system',
-      executable: PlatformBinding.instance.getCommandToRunEdge(),
+      executable: io.Directory(path.join(
+              edgeLauncher.launcherInstallationDir.path,
+              '${PlatformBinding.instance.getCommandToRunEdge()}'))
+          .path,
     );
   } else {
     infoLog.writeln('Unsupported version $requestedVersion.');
     throw UnimplementedError();
   }
 }
+
+class EdgeLauncher {
+  /// Url for downloading  `MicrosoftEdgeLauncher`.
+  ///
+  /// Only useful in Windows, hence not added to [PlatformBinding].
+  final String windowsEdgeLauncherDownloadUrl =
+      'https://github.com/MicrosoftEdge/edge-launcher/releases/download/1.2.0.0/MicrosoftEdgeLauncher.exe';
+
+  /// Path to the directory contains `MicrosoftEdgeLauncher.exe`.
+  io.Directory get launcherInstallationDir => io.Directory(
+        path.join(environment.webUiDartToolDir.path, 'microsoftedgelauncher'),
+      );
+
+  io.File get executable => io.File(
+      path.join(launcherInstallationDir.path, 'MicrosoftEdgeLauncher.exe'));
+
+  bool get isInstalled => executable.existsSync();
+
+  /// For running Edge from a batch script executable `MicrosoftEdgeLauncher`
```

Comment:
Which batch script are you referring to?

Reply:
The text copied from their github project "A simple command line exe to launch the Microsoft Edge browser at a URL. Handy if you need to launch Microsoft Edge from the command line or a batch script etc." 

On the other hand it is not entirely false since I'm also planning to write a batch script for windows (felt_windows maybe?) since the existing one doesn't' work. Here is the issue:  https://github.com/flutter/flutter/issues/48827

Reply:
Ah, I think they _literally_ mean the `*.bat` files, which are equivalent to `*.sh` files in *nix systems. Do we plan to use those?

Reply:
done.